src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1710,6 +1710,8 @@ class TransactionStatementSegment(BaseSegment):
             Ref.keyword("WORK", optional=True),
             Sequence("AND", Ref.keyword("NO", optional=True), "CHAIN", optional=True),
         ),
+        # BEGIN | END TRANSACTION
+        Sequence(OneOf("BEGIN", "END"), "TRANSACTION"),
```

Comment:
Should we make "TRANSACTION" optional? it is optional in Snowflake: https://docs.snowflake.com/en/sql-reference/sql/begin.html and in Postgres https://www.postgresql.org/docs/9.0/sql-begin.html

Reply:
good catch